src/main.cpp

```diff
@@ -6062,23 +6068,41 @@ bool ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vRecv, in
     }
 
 
-    else if (strCommand == NetMsgType::XPEDITEDREQUEST)
+    else if (strCommand == NetMsgType::XPEDITEDREQUEST && IsThinBlocksEnabled())
     {
+        if (!pfrom->ThinBlockCapable())
+        {
+            LOCK(cs_main);
+            Misbehaving(pfrom->GetId(), 100);
+            return error("XPEDITEDREQUEST message received from a non thinblock node, peer=%d", pfrom->GetId());
+        }
```

Comment:
let's be more tolerant here.  How about Misbehaving 1 just to stop lots of repeat requests.